File: 6086.cpp
#include <iostream>
#include <vector>
#include <queue>

#define MAX 701
#define INF 1000000

using namespace std;

int n, result;
int c[MAX][MAX], f[MAX][MAX], d[MAX];
vector<int> a[MAX];

void maxFlow(int start, int end) {
    while (1) {
        fill(d, d + MAX, -1);
        queue<int> q;
        q.push(start);
        while (!q.empty()) {
            int x = q.front();
            q.pop();
            for (int i = 0; i < a[x].size(); ++i) {
                int y = a[x][i];

                if (c[x][y] - f[x][y] > 0 && d[y] == -1) {
                    q.push(y);
                    d[y] = x;
                    if (y == end) break;
                }
            }
        }
        if (d[end] == -1) break;
        int flow = INF;
        for (int i = end; i != start; i = d[i]) {
            flow = min(flow, c[d[i]][i] - f[d[i]][i]);
        }
        for (int i = end; i != start; i = d[i]) {
            f[d[i]][i] += flow;
            f[i][d[i]] -= flow;
        }
        result += flow;
    }
}

int main() {
    cin >> n;

    for (int i = 0; i < n; ++i) {
        char x, y;
        int z;
        cin >> x >> y >> z;
        a[x].push_back(y);
        a[y].push_back(x);
        c[x][y] += z;
        c[y][x] += z;
    }

    maxFlow('A', 'Z');

    cout << result << endl;
}

File: 6086.py
from collections import deque

def char_to_int(c):
    if c.isupper():
        return ord(c) - ord('A')
    return ord(c) - ord('a') + 26

def max_flow(source, sink):
    total_flow = 0
    while True:
        parent = [-1] * 52
        queue = deque([source])
        parent[source] = source
        
        while queue and parent[sink] == -1:
            current = queue.popleft()
            for next_node in graph[current]:
                residual = capacity[current][next_node] - flow[current][next_node]
                if residual > 0 and parent[next_node] == -1:
                    parent[next_node] = current
                    queue.append(next_node)
        
        if parent[sink] == -1:
            break
        
        path_flow = float('inf')
        node = sink
        while node != source:
            parent_node = parent[node]
            path_flow = min(path_flow, capacity[parent_node][node] - flow[parent_node][node])
            node = parent_node
        
        node = sink
        while node != source:
            parent_node = parent[node]
            flow[parent_node][node] += path_flow
            flow[node][parent_node] -= path_flow
            node = parent_node
        
        total_flow += path_flow
    
    return total_flow

N = int(input())
graph = [[] for _ in range(52)]
capacity = [[0] * 52 for _ in range(52)]
flow = [[0] * 52 for _ in range(52)]

for _ in range(N):
    a, b, c = input().split()
    a, b = char_to_int(a), char_to_int(b)
    c = int(c)
    
    graph[a].append(b)
    graph[b].append(a)
    capacity[a][b] += c
    capacity[b][a] += c

source = char_to_int('A')
sink = char_to_int('Z')

print(max_flow(source, sink))